pkg/ddc/base/syncs.go

```diff
@@ -53,6 +55,10 @@ func (t *TemplateEngine) Sync(ctx cruntime.ReconcileRequestContext) (err error)
 	}
 
 	// 4. Update dataset
+	_, err = t.Implement.UpdateOnUFSChange()
```

Comment:
The logic of Sync() is changed, so the corresponding test logic should also be updated to pass unit test.